worlds/kh2/__init__.py

```diff
@@ -324,78 +435,98 @@ def starting_invo_verify(self):
                 self.item_quantity_dict[item] -= value
 
     def emblem_verify(self):
-        if self.luckyemblemamount < self.luckyemblemrequired:
+        """
+        Making sure lucky emblems have amount>=required.
+        """
+        if self.lucky_emblem_amount < self.lucky_emblem_required:
             logging.info(
-                    f"Lucky Emblem Amount {self.multiworld.LuckyEmblemsAmount[self.player].value} is less than required "
-                    f"{self.multiworld.LuckyEmblemsRequired[self.player].value} for player {self.multiworld.get_file_safe_player_name(self.player)}."
-                    f" Setting amount to {self.multiworld.LuckyEmblemsRequired[self.player].value}")
-            luckyemblemamount = max(self.luckyemblemamount, self.luckyemblemrequired)
-            self.multiworld.LuckyEmblemsAmount[self.player].value = luckyemblemamount
+                    f"Lucky Emblem Amount {self.options.LuckyEmblemsAmount.value} is less than required "
+                    f"{self.options.LuckyEmblemsRequired.value} for player {self.multiworld.get_file_safe_player_name(self.player)}."
+                    f" Setting amount to {self.options.LuckyEmblemsRequired.value}")
+            luckyemblemamount = max(self.lucky_emblem_amount, self.lucky_emblem_required)
+            self.options.LuckyEmblemsAmount.value = luckyemblemamount
 
-        self.item_quantity_dict[ItemName.LuckyEmblem] = self.multiworld.LuckyEmblemsAmount[self.player].value
+        self.item_quantity_dict[ItemName.LuckyEmblem] = self.options.LuckyEmblemsAmount.value
         # give this proof to unlock the final door once the player has the amount of lucky emblem required
         self.item_quantity_dict[ItemName.ProofofNonexistence] = 0
```

Comment:
Here too, it's better to just del the key instead of setting the value to 0 since it frees up memory unless you still need it in the dictionary for some reason.